src/components/Zoom/Zoom.js

```diff
@@ -90,7 +96,26 @@ function Zoom({
     [map],
   );
 
+  const zoomInDisabled = useMemo(() => {
+    return (
+      map.getView().getZoom() >=
+      map.getView().getConstrainedZoom(map.getView().getMaxZoom())
+    );
+  }, [currentZoom]);
+
+  const zoomOutDisabled = useMemo(() => {
+    return (
+      map.getView().getZoom() <=
+      map.getView().getConstrainedZoom(map.getView().getMinZoom())
+    );
+  }, [currentZoom]);
+
   useEffect(() => {
+    /* Trigger recalculation of disabled zooms */
+    map.on('moveend', () => {
```

Comment:
you have to unlisten this function in the function returned by the useEffect

Reply:
Done